ci/build/build-docker-images.py

```diff
@@ -886,6 +886,7 @@ def main(
         _configure_human_version()
     if (
         build_type in {HUMAN, MERGE, BUILDKITE, LOCAL}
+        or (build_type == PR and push_pr_build)
```

Comment:
synced offline, will add a new `build_type` as `RELEASE_PR` for PR runs on release for now.

in the mid/long term, we will move towards just building the containers on the release test pipeline if it does not exist, no matter if it is a pr or not.

the current "pr" dependency is just the base image for running the docker build, which is built from the CI PR pipeline. but the dependency itself is actually not necessary.